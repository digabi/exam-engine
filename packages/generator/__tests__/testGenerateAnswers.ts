import { GradingStructure } from '@digabi/exam-engine-core'
import { generateAnswers } from '@digabi/exam-engine-generator'

describe('generateAnswers()', () => {
  it('creates an answer for each text question in grading structure', () => {
    const gradingStructure: GradingStructure = {
      questions: [
        { type: 'text', id: 1, displayNumber: '1', maxScore: 10 },
        { type: 'text', id: 2, displayNumber: '2', maxScore: 10 }
      ]
    }
    expect(generateAnswers(gradingStructure)).toMatchSnapshot()
  })

  it('creates an answer for each choice question in grading structure', () => {
    const gradingStructure: GradingStructure = {
      questions: [
        {
          type: 'choicegroup',
          id: 1,
          displayNumber: '1',
          choices: [
            {
              type: 'choice',
              id: 2,
              displayNumber: '1.1',
              options: [
                { id: 3, correct: true, score: 2 },
                { id: 4, correct: false, score: 0 },
                { id: 5, correct: false, score: 0 }
              ]
            },
            {
              type: 'choice',
              id: 6,
              displayNumber: '1.2',
              options: [
                { id: 7, correct: false, score: 2 },
                { id: 8, correct: true, score: 0 },
                { id: 9, correct: false, score: 0 }
              ]
            },
            {
              type: 'choice',
              id: 10,
              displayNumber: '1.3',
              options: [
                { id: 11, correct: false, score: 0 },
                { id: 12, correct: false, score: 0 },
                { id: 13, correct: true, score: 2 }
              ]
            }
          ]
        }
      ]
    }

    expect(generateAnswers(gradingStructure)).toMatchSnapshot()
  })
})
