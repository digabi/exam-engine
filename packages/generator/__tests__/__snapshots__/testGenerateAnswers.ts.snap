// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateAnswers() creates an answer for each choice question in grading structure 1`] = `
[
  {
    "questionId": 2,
    "type": "choice",
    "value": "3",
  },
  {
    "questionId": 6,
    "type": "choice",
    "value": "8",
  },
  {
    "questionId": 10,
    "type": "choice",
    "value": "13",
  },
]
`;

exports[`generateAnswers() creates an answer for each text question in grading structure 1`] = `
[
  {
    "characterCount": 17,
    "questionId": 1,
    "type": "text",
    "value": "Answer to question 1",
  },
  {
    "characterCount": 17,
    "questionId": 2,
    "type": "text",
    "value": "Answer to question 2",
  },
]
`;
